sphinx/domains/c.py

```diff
@@ -203,7 +203,6 @@ def add_target_and_index(self, name: str, sig: str, signode: desc_signature) ->
         if targetname not in self.state.document.ids:
             signode['names'].append(targetname)
             signode['ids'].append(targetname)
-            signode['first'] = (not self.names)
```

Comment:
Before calling `add_target_and_index()`, `self.names` are always filled. So this assigns `False` always.